### PR TITLE
chore: updates due dates

### LINKED ISSUE #1057: Change pending updates due date according to vesting start date
Currently, the due dates of status updates for grants are defined by the end date of the governance proposal related to that grant. This means that if a proposal ended on July 9th, all the project update due dates are going to be on the 9th of each month.

This card proposes a **change to this logic** to align it with the new vesting contracts and overall grants program.

- Since the vesting contract are always initialized on the 1st day or the 15th day of the month, the **project update due dates should align with those dates**. This will also order projects over time and synchronize updates made by grantees through the Governance dApp at the end of half of the month.

If we implement this right, a grant that was funded the 1st day of the month will have their update deadline setted for the 1st day of the month and a grant that was funded the 15th day of the month will have their update deadlines setted for the 15th day of the month.

### PR DESCRIPTION
**Pull request by 1emu**
Closes #1057 

After manually testing, I noticed we are calculating the months for a vesting inaccurately.

For example, [this vesting ](https://vesting.decentraland.org/#/0x26d9a31b33179a856f9338644ec62b76cba15327) has a duration of 4 months and 25 days, but we consider it to last 4 months. 

I've included `moment.js` and a plugin that lets us calculate the exact difference between vesting dates, so we create the right amount of updates.

If preferred payment day is set for 15, updating a grant with the mentioned vesting of more than 4 months generates 5 updates:

```
vestingContractData {
  vestingStartAt: '2023-06-06T00:00:00.000Z',
  vestingFinishAt: '2023-10-31T00:00:00.000Z',
}

Update dueDate 2023-07-15T00:00:00.000Z
Update dueDate 2023-08-15T00:00:00.000Z
Update dueDate 2023-09-15T00:00:00.000Z
Update dueDate 2023-10-15T00:00:00.000Z
Update dueDate 2023-11-15T00:00:00.000Z
```

For a vesting of exactly 6 months, and a proposal with preffered payment on the 15th, it creates 6 updates:

```
vestingContractData {
  vestingStartAt: '2020-07-01 00:00:00z',
  vestingFinishAt: '2021-01-01 00:00:00z',
}

Update dueDate 2020-08-15T00:00:00.000Z
Update dueDate 2020-09-15T00:00:00.000Z
Update dueDate 2020-10-15T00:00:00.000Z
Update dueDate 2020-11-15T00:00:00.000Z
Update dueDate 2020-12-15T00:00:00.000Z
Update dueDate 2021-01-15T00:00:00.000Z
```

Update modal shows an error if fetching vesting data (or something along the way) fails.

image

image

### CODE REVIEW

File: src/components/Modal/ProposalModal.css [hunk 1]
```diff
@@ -16,13 +16,13 @@
...
 .ProposalModal .ProposalModal__Title .ui.header {
-  margin-bottom: .5rem;
+  margin-bottom: 0.5rem;
 }
 
 .ProposalModal .ui.button.cancel {
   margin-top: 8px;
   background-color: var(--background);
-  border: 1px solid rgba(115,110,125,.32);
+  border: 1px solid rgba(115, 110, 125, 0.32);
```
Comment (ncomerci) [inline src/components/Modal/ProposalModal.css:25]: --alpha-black-400

File: src/components/Modal/UpdateProposalStatusModal/UpdateProposalStatusModal.tsx [hunk 2]
```diff
@@ -27,6 +30,13 @@ type UpdateProposalState = {
   description: string
 }
 
+type UpdateProps = {
+  status: ProposalStatus
+  vesting_contract: string | null
+  enactingTx: string | null
```
Comment (ncomerci) [inline src/components/Modal/UpdateProposalStatusModal/UpdateProposalStatusModal.tsx:36]: camel case or snake case?

File: src/components/Error/ErrorMessage.tsx [hunk 3]
```diff
@@ -27,10 +29,12 @@ export default function ErrorMessage({ label, errorMessage }: Props) {
   }
 
   return (
-    <div className="ErrorMessage__Container">
-      <div className="ErrorMessage__Header">
-        <ErrorNotice />
-        <span>{label}</span>
+    <div className={classNames(className, 'ErrorMessage__Container')}>
```
Comment (andyesp) [inline src/components/Error/ErrorMessage.tsx:32]: Maybe className has to be after `ErrorMessage__Container` if we want to override something without `!important` here?

File: src/components/Error/ErrorMessage.css [hunk 4]
```diff
@@ -7,12 +7,23 @@
   border-radius: 6px;
 }
 
+.ErrorMessage__Header-vertical {
```
Comment (andyesp) [inline src/components/Error/ErrorMessage.css:10]: It should be `*--vertical` with an extra `-`

File: src/components/Modal/UpdateProposalStatusModal/UpdateProposalStatusModal.tsx [hunk 5]
```diff
@@ -60,30 +66,69 @@ const getPrimaryButtonTextKey = (status?: ProposalStatus | null) => {
...
+  proposalKey,
   className,
   ...props
 }: Props) {
   const t = useFormatMessage()
+  const isOneTimePaymentProposalTier = GrantTier.isOneTimePaymentTier(proposal?.configuration.tier)
+  const queryClient = useQueryClient()
+
+  const onClose = () => {
+    setError('')
```
Comment (andyesp) [inline src/components/Modal/UpdateProposalStatusModal/UpdateProposalStatusModal.tsx:83]: Should we also reset the form field values too?

File: src/components/Modal/UpdateProposalStatusModal/UpdateProposalStatusModal.tsx [hunk 6]
```diff
@@ -60,30 +66,69 @@ const getPrimaryButtonTextKey = (status?: ProposalStatus | null) => {
...
+            proposal.id,
+            status,
+            vesting_contract,
+            enactingTx,
+            description
+          )
+          onClose()
+          return updateProposal
+        } catch (err: any) {
+          console.error(err, { ...err })
```
Comment (andyesp) [inline src/components/Modal/UpdateProposalStatusModal/UpdateProposalStatusModal.tsx:111]: I'd remove this console.error

File: src/components/Modal/UpdateProposalStatusModal/UpdateProposalStatusModal.tsx [hunk 7]
```diff
@@ -60,30 +66,69 @@ const getPrimaryButtonTextKey = (status?: ProposalStatus | null) => {
...
+          setError(err.body?.error || err.message)
+        }
+      }
+    },
+    onSuccess: (proposal) => {
+      if (proposal) {
+        queryClient.setQueryData([proposalKey], proposal)
+      }
+    },
+    mutationKey: ['updatingProposal'],
```
Comment (andyesp) [inline src/components/Modal/UpdateProposalStatusModal/UpdateProposalStatusModal.tsx:121]: Should we add the proposal id to the key? 🤔

File: src/entities/Updates/model.ts [hunk 8]
```diff
@@ -1,29 +1,56 @@
...
+    if (proposalId.length < 0) throw new Error('Unable to create updates for empty proposal id')
+
     const now = new Date()
+    const updatesQuantity = vestingDates.durationInMonths
+    const firstUpdateStartingDate = this.getFirstUpdateStartingDate(
+      vestingDates.vestingStartAt,
+      preferredVestingStartDate
+    )
+
+    await UpdateModel.delete<UpdateAttributes>({ proposal_id: proposalId, status: UpdateStatus.Pending })
```
Comment (andyesp) [inline src/entities/Updates/model.ts:29]: Isn't this deleting just one record?

Comment (1emu) [reply src/entities/Updates/model.ts:29]: It deletes all records that match those conditions

File: src/components/Modal/UpdateProposalStatusModal/UpdateProposalStatusModal.tsx (hunk 5 above)
Comment (1emu) [reply src/components/Modal/UpdateProposalStatusModal/UpdateProposalStatusModal.tsx:83]: I don't think it's necessary

File: src/clients/VestingData.ts [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+  vestingStartAt: string
+  vestingFinishAt: string
+}
+
+function toISOString(seconds: number) {
+  return new Date(seconds * 1000).toISOString()
+}
+
+function getVestingDates(contractStart: number, contractEndsTimestamp: number) {
+  const vestingStartAt = toISOString(contractStart)
```
Comment (andyesp) [inline src/clients/VestingData.ts:19]: Do we need this as iso string or this function could already return a `Date`?

**Review by andyesp — APPROVED** [review]

**Review by ncomerci — APPROVED** [review]

**Pull request merged**